### PR TITLE
feat: skip ipfs parts on drops and update metadata v2

### LINKED ISSUE #10001: Unable to mint Tilescapes: Stuck on loading to ipfs
I wanted to mint 6 pieces of [Tilescapes](https://kodadot.xyz/ahp/drops/tilescape) all at once. I selected number 6 and clicked on Mint. I have enough funds on ahp. I got stuck on loading to ipfs. I waited 5 minutes, but nothing happened. It was supposed to take 15 seconds:

https://github.com/kodadot/nft-gallery/assets/104380776/29178a33-3375-41c6-a0ba-7903a4fdd2cc

I closed the modal with x after that. It looks like it's still trying to do something:

Screenshot 2024-04-05 at 12 08 37

I waited another 5 minutes. Nothing happened.

My address: 1NFfEH3yspdEgLnhZ5QWgb7B2z5LNCAE3HUo5nDxF5MqTcj

I refreshed the page and tried again - but I got the same issue. I'm just not able to mint guys...

### LINKED ISSUE #10047: CORS when uploading to IPFS on Drops
Error prevents user to advance from upload to ipfs step: 

image

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [Koda - Generative Art Marketplace](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Feature

## Context

- [x] Closes #10001
- [x] Closes #10047

related PR https://github.com/kodadot/private-workers/pull/116

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p)

### CODE REVIEW

**Comment by prury-temporary** [comment]
Tried this one and I'm still stuck on uploading to IPFS for 5+ minutes
- was able to reproduce with 8 items
- you must have the full funds on assethub, because if you need to teleport from polkadot it works properly and doesn't show upload to IPFS

presch-upload

changes-requested

**Comment by preschian** [comment]
Checking, that text should be removed

**Comment by preschian** [comment]
> Tried this one and I'm still stuck on uploading to IPFS for 5+ minutes

@prury-temporary, could you check again? I just changed the error text like this.



> you must have the full funds on assethub, because if you need to teleport from polkadot it works properly and doesn't show upload to IPFS

Can you rephrase this one? Is it related to auto-teleport?

File: services/fxart.ts [hunk 1]
```diff
@@ -143,15 +121,27 @@ export const allocateClaim = async (body, id) => {
...
+    'https://fxart-beta.kodadot.workers.dev/metadata/v2/json',
+  )
+  metadataUrl.searchParams.set('chain', chain)
+  metadataUrl.searchParams.set('collection', collection)
+  metadataUrl.searchParams.set('hash', hash)
+
+  return metadataUrl.toString()
+}
+
+export const updateMetadata = async ({ chain, collection, nft, sn, hash }) => {
```
Comment (vikiival) [inline services/fxart.ts:135]: can we please use the `dyndata` format for v2?

EVM uri is resolved by baseURI + sn


https://github.com/OpenZeppelin/openzeppelin-contracts/blob/11dc5e3809ebe07d5405fe524385cbe4f890a08b/contracts/token/ERC721/ERC721.sol#L88-L93

**Comment by prury** [comment]
@preschian found something: 

to reproduce: generate variations, close modal, generate variations again

[Gravação de tela de 23-04-2024 10:25:16.webm](https://github.com/kodadot/nft-gallery/assets/36627808/6f1c262a-b75c-477c-a186-135eb80fbd98)

File: services/fxart.ts (hunk 1 above)
Comment (preschian) [reply services/fxart.ts:135]: Let's use both of them first.
```
evm -> `dyndata`
assethub -> `fxart`
```
Once both of them are stable, let's merge `assethub` into `dyndata`

I still have some work to do on dynamic data on fxart: validation and endpoint migrator (since we moved to find the index to the frontend side, there is a case when the user can accidentally miss the update metadata endpoint). Once the requirements are finalised on `fxart`, let's move that to `dyndata`.

I think even on `dyndata`, it is better to separate them with prefix endpoints to reduce complexity in one endpoint. For example:
```
dyndata.dev/v1/metadata/evm
dyndata.dev/v1/metadata/assethub
```

**Comment by vikiival** [comment]
cc @Jarsen136, @hassnian  can I ask for the review

**Review by Jarsen136 — APPROVED** [review]

**Review by hassnian — APPROVED** [review]
otherwise LGTM

File: composables/drop/massmint/useDropMassMint.ts [hunk 2]
```diff
@@ -1,25 +1,23 @@
...
-import useDropMassMintUploader from './useDropMassMintUploader'
-import { useCollectionEntity } from '../useGenerativeDropMint'
 
 export type MassMintNFT = Omit<ToMintNft, 'priceUSD'> & {
   imageDataPayload?: ImageDataPayload
   metadata?: string
   hash: string
-  sn?: number
+  sn?: number // nft id
+  index?: number // serial numbers
```
[the remark this answers is not in the document]
Comment (vikiival) [reply composables/drop/massmint/useDropMassMint.ts:12]: Id is collection + sn

[the remark this answers is not in the document]
Comment (preschian) [reply composables/drop/massmint/useDropMassMint.ts:12]: Agree, at first, I want to do that also. But it seems like there are so many places that need to check. Currently, I'm cleaning some parts, e.g., the free drops component in the other branch. I will try to do that in this branch instead

Screenshot 2024-04-25 at 02 35 12

[the remark this answers is not in the document]
Comment (vikiival) [reply composables/drop/massmint/useDropMassMint.ts:12]: Please try to follow the indexer schema. If there is something that needs to be updated He can update the schema

**Comment by preschian** [comment]
@prury @vikiival do we have deployment to prod?
